traits/tests/test_has_traits.py

```diff
@@ -366,6 +366,18 @@ def test__trait_set_inited(self):
 
         self.assertTrue(foo.traits_inited())
 
+    def test___dir__(self):
```

Comment:
There is already a `test_dir` test in `test_get_traits.py`; that should probably be merged with this one (it tests that uninitialized traits and non-trait class variables are reported correctly).